src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs

```diff
@@ -35,36 +35,47 @@ pub struct DedupPkCellBasedRowDeserializer<Desc: Deref<Target = ColumnDescMappin
     pk_to_row_mapping: Vec<Option<usize>>,
 }
 
-impl<Desc: Deref<Target = ColumnDescMapping>> DedupPkCellBasedRowDeserializer<Desc> {
-    /// Create a [`DedupPkCellBasedRowDeserializer`]
-    /// to decode cell based row with dedup pk encoding.
-    /// TODO: Refactor args. Creating `DedupPkCellBasedRowDeserializer` should not have overhead.
-    pub fn new(column_mapping: Desc, pk_descs: &[OrderedColumnDesc]) -> Self {
-        let (pk_data_types, pk_order_types) = pk_descs
-            .iter()
-            .map(|ordered_desc| {
-                (
-                    ordered_desc.column_desc.data_type.clone(),
-                    ordered_desc.order,
-                )
-            })
-            .unzip();
-        let pk_deserializer = OrderedRowDeserializer::new(pk_data_types, pk_order_types);
-
-        let pk_to_row_mapping = pk_descs
-            .iter()
-            .map(|d| {
-                let column_desc = &d.column_desc;
-                if column_desc.data_type.mem_cmp_eq_value_enc() {
-                    column_mapping
-                        .get(column_desc.column_id)
-                        .map(|(_, index)| index)
-                } else {
-                    None
-                }
-            })
-            .collect();
+pub fn create_pk_deserializer_from_pk_descs(
```

Comment:
Add some doc or example on pub function?
Can not get quickly that why this change is necessary.

Reply:
Good catch, updated it.